### DISCUSSION TITLE
An auto-delete queue is not deleted, UI says it has 0 consumers

### DISCUSSION THREAD

**Discussion by aquishix** [rabbitmq/rabbitmq-server#10474] [Other]
I've been debugging my own code that is supposed to cause my consumer wrapper to correctly reconnect to the broker, recreate its queue, and rebind that queue to a given exchange.  The queue is set explicitly to auto-delete in the queueDeclare invocation.  Yet I see the queue on my broker web UI with **AD** showing, and no consumers on it.  It has been in this state for several minutes.  Thus it appears to be a bug, despite my inability to tell you any more information about how to recreate the conditions.  I have attached the RabbitMQ log and several screenshots to this bug report in the hopes that it'll help.

This is when using RabbitMQ 3.7.9, but I haven't seen any existing issues, resolved or unresolved, that pertain to this behavior -- so I'm assuming that if this really is a bug, it's present in later versions.

![screenshot_0](https://user-images.githubusercontent.com/1828709/53916114-f2536f80-4026-11e9-978c-8a15ba7a4f66.png)

![screenshot_1](https://user-images.githubusercontent.com/1828709/53916140-07300300-4027-11e9-9678-7da3d5b1be41.png)

[rabbit.log](https://github.com/rabbitmq/rabbitmq-server/files/2938457/rabbit.log)

**Comment by michaelklishin** [comment]
Thank you for your time.

Team RabbitMQ uses GitHub issues for specific actionable items engineers can work on. GitHub issues are not used for questions, investigations, root cause analysis, discussions of potential issues, etc (as defined by this team).

We get at least a dozen of questions through various venues every single day, often light on details.
At that rate GitHub issues can very quickly turn into a something impossible to navigate and make sense of even for our team. Because GitHub is a tool our team uses heavily nearly every day, the signal/noise ratio of issues is something we care about a lot.

Please post this to [rabbitmq-users](https://groups.google.com/forum/#!forum/rabbitmq-users).

Thank you.

**Comment by michaelklishin** [comment]
I'm pretty certain that the queue never had any consumers. An auto-delete queue that [never had a consumer will not be deleted](http://www.rabbitmq.com/queues.html#properties).

If you have an executable way to reproduce, please start a mailing list thread.

**Comment by michaelklishin** [comment]
[rabbitmq-event-exchange](https://github.com/rabbitmq/rabbitmq-event-exchange) allows applications to monitor connection, channel, queue and consumer events. Since it require using a regular RabbitMQ queue and consumers, connection recovery should be tested by closing [other]  connections with `rabbitmqctl` or HTTP API. Stopping and starting the node won't work very well.

**Comment by spasaria** [comment]
I have the same problem! Did you find out anything about it that can help me? 
Thank you.

**Comment by ovolkovsky** [comment]
I think you have to set x-expires on the queue, and it will be deleted

**Comment by michaelklishin** [comment]
> I have the same problem! Did you find out anything about it that can help me? Thank you.

Not with a single line problem definition. https://github.com/rabbitmq/rabbitmq-server/issues/1909#issuecomment-470300030 is the best guess, and it is still my best guess today.

Instead of posting comments on a random issue from 2019, consider starting a new discussion with relevant details.

**Comment by michaelklishin** [comment]
> I think you have to set x-expires on the queue, and it will be deleted

If my hypothesis above is correct, then [setting TTL on such queues](https://rabbitmq.com/ttl.html) that never have any consumers come online will indeed help.

**Reply by kwerie** [reply]
I have tried setting a TTL/expiry on a (delay) queue with `x-expires` (and `x-message-ttl` so the messages will be re-routed when the message TTL expires) but the queue is being deleted even though it contains messages. Do you have any ideas for this scenario?
